aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -145,7 +206,12 @@ module aptos_framework::coin {
 
     /// Returns the amount of coin in existence.
     public fun supply<CoinType>(): Option<u128> acquires CoinInfo {
-        borrow_global<CoinInfo<CoinType>>(coin_address<CoinType>()).supply
+        let supply = &borrow_global<CoinInfo<CoinType>>(coin_address<CoinType>()).supply;
+        if (option::is_some(supply)) {
+            option::some(read(option::borrow(supply)))
```

Comment:
Why not move option::borrow into read?

Reply:
I think the problem here is that we have Option<Option<Aggregator>> so we need to have this `borrow`